src/revivalpmmp/pureentities/entity/WalkingEntity.php

```diff
@@ -222,49 +223,80 @@ protected function checkJump($dx, $dz){
-			$upperBlock = $this->getLevel()->getBlock($blockingBlock->add(0, 1, 0));
-			$secondUpperBlock = $this->getLevel()->getBlock($blockingBlock->add(0, 2, 0));
-			PureEntities::logOutput("$this: checkJump(): block in front is $blockingBlock, upperBlock is $upperBlock, second Upper block is $secondUpperBlock");
-			// check if we can get through the upper of the block directly in front of the entity
-			if($upperBlock->canPassThrough() && $secondUpperBlock->canPassThrough()){
-				if($blockingBlock instanceof Fence || $blockingBlock instanceof FenceGate){ // cannot pass fence or fence gate ...
-					$this->motion->y = $this->gravity;
-					PureEntities::logOutput("$this: checkJump(): found fence or fence gate!", PureEntities::DEBUG);
-				}else if($blockingBlock instanceof StoneSlab or $blockingBlock instanceof Stair){ // on stairs entities shouldn't jump THAT high
-					$this->motion->y = $this->gravity * 4;
-					PureEntities::logOutput("$this: checkJump(): found slab or stair!", PureEntities::DEBUG);
-				}else if($this->motion->y < ($this->gravity * 3.2)){ // Magic
-					PureEntities::logOutput("$this: checkJump(): set motion to gravity * 3.2!", PureEntities::DEBUG);
-					$this->motion->y = $this->gravity * 3.2;
-				}else{
-					PureEntities::logOutput("$this: checkJump(): nothing else!", PureEntities::DEBUG);
-					$this->motion->y += $this->gravity * 0.25;
-				}
-				return true;
-			}elseif(!$upperBlock->canPassThrough()){
-				PureEntities::logOutput("$this: checkJump(): cannot pass through the upper blocks!", PureEntities::DEBUG);
-				$this->yaw = $this->getYaw() + mt_rand(-120, 120) / 10;
+			
+		
+
+			$blockBox = $block->getCollisionBoxes()[0] ?? null;
+			$blockBoxDiff = $blockBox === null ? 0 : $blockBox->maxY - $blockBox->minY;
+			//Do the same for the upper block, id there is any collisionBox, consider as a block
+			//If i jump and have 2 slab or stair, so i cannot jump over these 2
+			PureEntities::logOutput("$this: CollisionBlock : $block --Height: $blockBoxDiff-- Pos: [x:" . $block->x . "] [y:" . $block->y . "] [z:" . $block->z . "]");
+			$upperblock = $this->getLevel()->getBlock($block->add(0, 1, 0));
+			if (!empty($upperblock->getCollisionBoxes())) {
+				$blockBoxDiff = $blockBoxDiff + 1;
+				PureEntities::logOutput("$this: CollisionBlock - upperBlock Detected - New Height($blockBoxDiff) : $upperblock - Pos: [x:" . $upperblock->x . "] [y:" . $upperblock->y . "] [z:" . $upperblock->z . "]");
 			}
-		}else{
-			PureEntities::logOutput("$this: checkJump(): no need to jump. Block can be passed! [canPassThrough:" . $blockingBlock->canPassThrough() . "] " .
-				"[jumpHeight:" . $this->getMaxJumpHeight() . "] [checkedBlock:" . $blockingBlock . "]", PureEntities::DEBUG);
+			if ($blockBoxDiff > $highestBlock) $highestBlock = $blockBoxDiff;
+			if ($lowestBlock == null) $lowestBlock = $blockBoxDiff;
+			elseif ($lowestBlock > $blockBoxDiff) $lowestBlock = $blockBoxDiff;
```

Comment:
Each of these if/elseif statements should have a body with curling braces.  This will maintain code style.

Reply:
Should be corrected too